airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/test_include_deleted.py

```diff
@@ -0,0 +1,165 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+from unittest import TestCase
+
+from airbyte_cdk.test.entrypoint_wrapper import EntrypointOutput
+from airbyte_cdk.test.mock_http import HttpMocker
+from airbyte_protocol.models import SyncMode
+
+from .config import ACCOUNT_ID, ConfigBuilder
+from .request_builder import get_account_request, get_ad_sets_request, get_ads_request, get_campaigns_request
+from .response_builder import get_account_response, get_ad_sets_response, get_ads_response, get_campaigns_response
+from .utils import config, read_output
+
+
+class TestIncludeDeleted(TestCase):
+    account_id = ACCOUNT_ID
+    filter_statuses_flag = "filter_statuses"
+    statuses = ["ACTIVE", "ARCHIVED"]
+
+    @staticmethod
+    def _read(config_: ConfigBuilder, stream_name: str, expecting_exception: bool = False) -> EntrypointOutput:
+        return read_output(
+            config_builder=config_,
+            stream_name=stream_name,
+            sync_mode=SyncMode.incremental,
+            expecting_exception=expecting_exception,
+        )
+
+    @HttpMocker()
+    def test_ads_stream(self, http_mocker: HttpMocker):
+        http_mocker.get(
+            get_account_request().build(),
+            get_account_response(),
+        )
+        filters = (
+            '[{"field":"ad.effective_status","operator":"IN","value":["ACTIVE","ARCHIVED"]},'
+            '{"field":"ad.updated_time","operator":"GREATER_THAN","value":1672531200}]'
+        )
+        fields = [
+            "bid_type",
+            "account_id",
+            "campaign_id",
+            "adset_id",
+            "adlabels",
+            "bid_amount",
+            "bid_info",
+            "status",
+            "creative",
+            "id",
+            "updated_time",
+            "created_time",
+            "name",
+            "targeting",
+            "effective_status",
+            "last_updated_by_app_id",
+            "recommendations",
+            "source_ad_id",
+            "tracking_specs",
+            "conversion_specs",
+        ]
+
+        http_mocker.get(
+            get_ads_request().with_limit(100).with_filtering(filters).with_fields(fields).with_summary().build(),
+            get_ads_response(),
+        )
+
+        output = self._read(config().with_ad_statuses(self.statuses), "ads")
```

Comment:
With `create_response_builder`  added, `with_record` could be applied here so it is more clear why one record is expected

Reply:
updated